### ISSUE TITLE
Allow components to override object type and location for the purposes of naming using config

### ISSUE THREAD

**Issue by todderasesareddot** [fabric-engine/Kraken#455]
We need more flexibility in control naming.  Allow metaData on sceneItems to allow for the override of the component location (e.g. if mouth component location is "M" then individual controls can have location overrides for "L" and "R").   Also allow the override of "type" so far as the config "type" is concerned (e.g. change a control from "Control" to "MouthControl").  The override type does not have to be a real class, it only need exist in the config.  If the type does not exist in the config, the override will be ignored and the real class type used.

Example metaData:
{"altLocation": "R", "altType": "MouthControl"}

**Comment by EricTRocks** [comment]
Yes this is a great idea. If you have a prototype, push to a branch so we can review and test.

**Comment by todderasesareddot** [comment]
Just did ;)
